core/commands/filestore.go

```diff
@@ -0,0 +1,139 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+
+	//ds "github.com/ipfs/go-datastore"
+	//bs "github.com/ipfs/go-ipfs/blocks/blockstore"
+	//butil "github.com/ipfs/go-ipfs/blocks/blockstore/util"
+	cmds "github.com/ipfs/go-ipfs/commands"
+	//cli "github.com/ipfs/go-ipfs/commands/cli"
+	//files "github.com/ipfs/go-ipfs/commands/files"
+	"github.com/ipfs/go-ipfs/core"
+	"github.com/ipfs/go-ipfs/filestore"
+	//"github.com/ipfs/go-ipfs/repo/fsrepo"
+	//cid "gx/ipfs/QmXfiyr2RWEXpVDdaYnD2HNiBk6UBddsvEP4RPfXb6nGqY/go-cid"
+	u "gx/ipfs/Qmb912gdngC1UWwTkhuW8knyRbcWeu5kqkxBpveLmW8bSr/go-ipfs-util"
+)
+
+var FileStoreCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Interact with filestore objects.",
+	},
+	Subcommands: map[string]*cmds.Command{
+		"ls":     lsFileStore,
+		"verify": verifyFileStore,
+	},
+}
+
+var lsFileStore = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "List objects in filestore.",
+	},
+	Run: func(req cmds.Request, res cmds.Response) {
+		_, fs, err := getFilestore(req)
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+		next, err := filestore.ListAll(fs)
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+		out := listResAsChan(next, req.Context())
+		res.SetOutput(out)
+	},
+	PostRun: func(req cmds.Request, res cmds.Response) {
+		if res.Error() != nil {
+			return
+		}
+		outChan, ok := res.Output().(<-chan interface{})
```

Comment:
This should be able to go in the marshaler. You can write to stderr and stdout in the marshaler the same way you can in the PostRun. If we do weird things like this in the post run then it breaks the API expectations in a weird way.